collator/src/lib.rs

```diff
@@ -345,6 +349,8 @@ fn run_collator_node<S, P, Extrinsic>(
 		spawner.clone(),
 	));
 
+	let build_parachain_context = build_parachain();
```

Comment:
why is it better to do this lazily?

Reply:
It's not better. It's no choice. Polkadot has 4 different types of service and you can't return a common type in Rust

Reply:
I don't follow. I see that you are branching based on the type of service in `start_collator`, but the `build_parachain` parameter does not vary. So why does it need to be lazy?

Reply:
Yes I just realized this night I was telling crap :confounded: 

The reason for this was the hope to replace the CollatorBuilder. See [here](https://github.com/paritytech/cumulus/issues/44#issuecomment-583317236) for the discussion. Using a ParachainContext directly would be great but unfortunately I couldn't do it because of these 4 different types of service and the limitation.

I can totally remove the "builder" closure and keep only the CollatorBuilder *OR* I can remove the CollatorBuilder entirely and keep only the closure. cc @bkchr ? (But I must keep one: I cannot remove the need of a builder of some sort. Otherwise I need to duplicate some code like I did [here](https://github.com/paritytech/cumulus/pull/53#discussion_r376309209))

In my opinion, at this point, I think it would simplify a bit to remove the collator builder as this is just a fancy closure in the end.

Reply:
(Actually I couldn't figure out how to use only a closure so I will keep the BuildParachainContext/CollatorBuilder for now)